infrastructure/logger/logger.go

```diff
@@ -26,199 +27,103 @@ var (
-	muxxLog = BackendLog.Logger("MUXX")
-	grpcLog = BackendLog.Logger("GRPC")
-	p2psLog = BackendLog.Logger("P2PS")
-	ntarLog = BackendLog.Logger("NTAR")
-	dnssLog = BackendLog.Logger("DNSS")
-	snvrLog = BackendLog.Logger("SNVR")
-	wsvcLog = BackendLog.Logger("WSVC")
-	reacLog = BackendLog.Logger("REAC")
-	prnmLog = BackendLog.Logger("PRNM")
-	blvlLog = BackendLog.Logger("BLVL")
+	// subsystemLoggers maps each subsystem identifier to its associated logger.
+	subsystemLoggers      = make(map[string]*Logger)
+	subsystemLoggersMutex sync.Mutex
 )
 
-// SubsystemTags is an enum of all sub system tags
-var SubsystemTags = struct {
-	ADXR,
-	AMGR,
-	CMGR,
-	KSDB,
-	KASD,
-	BDAG,
-	CNFG,
-	DISC,
-	INDX,
-	MINR,
-	PEER,
-	RPCS,
-	RPCC,
-	SCRP,
-	SRVR,
-	SYNC,
-	TXMP,
-	UTIL,
-	PROF,
-	PROT,
-	MUXX,
-	GRPC,
-	P2PS,
-	NTAR,
-	DNSS,
-	SNVR,
-	WSVC,
-	REAC,
-	PRNM,
-	BLVL string
-}{
-	ADXR: "ADXR",
-	AMGR: "AMGR",
-	CMGR: "CMGR",
-	KSDB: "KSDB",
-	KASD: "KASD",
-	BDAG: "BDAG",
-	CNFG: "CNFG",
-	DISC: "DISC",
-	INDX: "INDX",
-	MINR: "MINR",
-	PEER: "PEER",
-	RPCS: "RPCS",
-	RPCC: "RPCC",
-	SCRP: "SCRP",
-	SRVR: "SRVR",
-	SYNC: "SYNC",
-	TXMP: "TXMP",
-	UTIL: "UTIL",
-	PROF: "PROF",
-	PROT: "PROT",
-	MUXX: "MUXX",
-	GRPC: "GRPC",
-	P2PS: "P2PS",
-	NTAR: "NTAR",
-	DNSS: "DNSS",
-	SNVR: "SNVR",
-	WSVC: "WSVC",
-	REAC: "REAC",
-	PRNM: "PRNM",
-	BLVL: "BLVL",
+// RegisterSubSystem Registers a new subsystem logger, should be called in a global variable,
+// returns the existing one if the subsystem is already registered
+func RegisterSubSystem(subsystem string) *Logger {
+	subsystemLoggersMutex.Lock()
+	defer subsystemLoggersMutex.Unlock()
+	logger, exists := subsystemLoggers[subsystem]
+	if !exists {
+		logger = BackendLog.Logger(subsystem)
+		subsystemLoggers[subsystem] = logger
+	}
+	return logger
 }
 
-// subsystemLoggers maps each subsystem identifier to its associated logger.
-var subsystemLoggers = map[string]*Logger{
-	SubsystemTags.ADXR: adxrLog,
-	SubsystemTags.AMGR: amgrLog,
-	SubsystemTags.CMGR: cmgrLog,
-	SubsystemTags.KSDB: ksdbLog,
-	SubsystemTags.KASD: kasdLog,
-	SubsystemTags.BDAG: bdagLog,
-	SubsystemTags.CNFG: cnfgLog,
-	SubsystemTags.DISC: discLog,
-	SubsystemTags.INDX: indxLog,
-	SubsystemTags.MINR: minrLog,
-	SubsystemTags.PEER: peerLog,
-	SubsystemTags.RPCS: rpcsLog,
-	SubsystemTags.RPCC: rpccLog,
-	SubsystemTags.SCRP: scrpLog,
-	SubsystemTags.SRVR: srvrLog,
-	SubsystemTags.SYNC: syncLog,
-	SubsystemTags.TXMP: txmpLog,
-	SubsystemTags.UTIL: utilLog,
-	SubsystemTags.PROF: profLog,
-	SubsystemTags.PROT: protLog,
-	SubsystemTags.MUXX: muxxLog,
-	SubsystemTags.GRPC: grpcLog,
-	SubsystemTags.P2PS: p2psLog,
-	SubsystemTags.NTAR: ntarLog,
-	SubsystemTags.DNSS: dnssLog,
-	SubsystemTags.SNVR: snvrLog,
-	SubsystemTags.WSVC: wsvcLog,
-	SubsystemTags.REAC: reacLog,
-	SubsystemTags.PRNM: prnmLog,
-	SubsystemTags.BLVL: blvlLog,
+// InitLogStdoutOnly attaches stdout to the backend log and starts the logger.
+func InitLogStdoutOnly(logLevel Level) {
```

Comment:
The word "Only" is somewhat misleading here... 
It's `InitLogStdoutPlusAllOtherBackendsThatIHaveAlreadyInited` (As can be seen when this is called from InitLog).

But seriously - I'd rename to `InitLogStdout`